### PR TITLE
Rails Blog Project: Milestone 8 System RSpec Testing

### PR DESCRIPTION
**Pull request by Profsain**
### I completed this part of the project by:
- Fixing N+1 loading problem
- Created integrated test for User page
- Created integrated test for Post page
- Push all changes to GitHub and Make a pull request
-

### CODE REVIEW

File: spec/system/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    end
+  end
+
+  describe 'Show' do
+    before(:all) do
+      @user = User.first
+      @post = @user.posts.first
+    end
+  end
+end
```
Comment (medaminedev66) [inline spec/system/posts_spec.rb:42]: - [ ] All the tests fail except one. Please ensure all the tests pass successfully:
image
image

- If you see that I'm missing something or something is not 100% clear, please reach out to me on slack or leave comments in the PR thread.
Remember to tag me in your question so I can receive the notification.

File: spec/system/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    it 'shows the last three posts' do
+      visit(user_path(@user))
+
+      @user.fetch_recent_posts.each do |post|
+        expect(page).to have_text(post.title)
+        expect(page).to have_text(post.text)
+        expect(page).to have_link(href: user_post_path(@user.id, post.id))
+      end
+    end
+  end
```
Comment (medaminedev66) [inline spec/system/users_spec.rb:65]: - [ ] For the users show page, kindly add:
  - Tests that assert you can see a button that lets you view all of a user's posts.
  - Tests that redirects to the user's post's index page when you click on a **see all posts**.

File: spec/system/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      visit user_posts_path(@user)
+
+      @posts.each do |post|
+        expect(page).to have_text(post.title)
+        expect(page).to have_text(post.text)
+        expect(page).to have_text(post.comments_counter)
+        expect(page).to have_text(post.likes_counter)
+      end
+    end
+
```
Comment (medaminedev66) [inline spec/system/posts_spec.rb:26]: - [ ] For the posts/index page, please add:
* Tests that asserts you can see the first comments on a post.
* Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
* Tests that redirects to the post's show page when you click on a post.

File: spec/system/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      expect(page).to have_text("#{@user.posts_counter} post") if @user.posts_counter > 0
+    end
+  end
+
+  describe 'Show' do
+    before(:all) do
+      @user = User.first
+      @post = @user.posts.first
+    end
+  end
```
Comment (medaminedev66) [inline spec/system/posts_spec.rb:41]: - [ ] For the posts/show page, Kindly add:
* Tests that asserts you can see the post's title.
* Tests that asserts you can see who wrote the post.
* Tests that asserts you can see how many comments it has.
* Tests that asserts you can see how many likes it has.
* Tests that asserts you can see the post body.
* Tests that asserts you can see the username of each commentor.
* Tests that asserts you can see the comment each commentor left.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @Profsain  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional.
- There are no linter errors.
- You have a descriptive PR.
- Your code is clean

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/system/posts_spec.rb (hunk 1 above)
Comment (Profsain) [reply spec/system/posts_spec.rb:42]: Thank you for the feedback. When I run the test all the test passes.
I currently have 25 test cases in total.
sys

File: spec/system/users_spec.rb (hunk 2 above)
Comment (Profsain) [reply spec/system/users_spec.rb:65]: I will implement the test required.

File: spec/system/posts_spec.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        expect(page).to have_text(post.likes_counter)
+      end
+    end
+
+    it 'shows the user name, photo, and number of posts' do
+      visit user_posts_path(@user)
+
+      expect(page).to have_text(@user.name)
+      expect(page).to have_css("img[src*=\"#{@user.photo}\"]")
+      expect(page).to have_text("#{@user.posts_counter} post") if @user.posts_counter > 0
```
Comment (julie-ify) [inline spec/system/posts_spec.rb:32]: - [ ] This rspec expectation is failing. Kindly fix it. From the error message you need to format from "1 post" to "post: 1".

  ```
  "post: #{@user.posts_counter}"
  ```

  ffn

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Profsain,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/system/posts_spec.rb (hunk 5 above)
Comment (Profsain) [reply spec/system/posts_spec.rb:32]: Thank you for your feedback.
I will correct it.

**Pull request merged**